### ISSUE TITLE
Roadmap to maturity

### ISSUE THREAD

**Issue by almarklein** [pygfx/wgpu-py#437]
This is a meta-issue that lists the needed work to get to a 1.0 level maturity.
Note that we cannot call the API "stable" until the WebGPU spec is final.

General:
  * Ongoing work to keep up with changes to the WebGPU spec.
  * Ongoing work to keep up with changes to wgpu-native.
 
API improvements / changes:
  * #391
  * #425
  * #394 

API completeness:
  * Render bundles.
  * Occlusion queries.
  * Debug markers.
  * A bunch of optional arguments in webGPU methods that wgpu-native does not expose yet.

Testing:
  * Units tests to cover 100% of the API.
  * More examples, as coverage, but also to demonstrate the use of certain features.

Documentation:
  * Improve the guide.
  * Tutorials.

**Comment by Vipitis** [comment]
I think a good goal will be to never panic and lose the python process. Always crash gracefully and raise some kind of Exception. There is some open issue for this in wgpu-native too

**Comment by Korijn** [comment]
> I think a good goal will be to never panic and lose the python process. Always crash gracefully and raise some kind of Exception. There is some open issue for this in wgpu-native too

You're right! We talked about this before with you. See https://github.com/pygfx/wgpu-py/issues/171
